Assets/Scripts/Models/Scheduler/ScheduledEvent.cs

```diff
@@ -260,6 +260,21 @@ public void Stop()
             RepeatsForever = false;
         }
 
+        public void SetCooldown(float newCooldown)
+        {
+            if (newCooldown < TimeToWait)
+            {
+                // Decrement TimeToWait by the amount of time that has passed since the last time this event fired
+                TimeToWait -= Cooldown - TimeToWait;
+                if (TimeToWait < 0)
+                {
+                    TimeToWait = 0;
```

Comment:
Should this fire the event? Maybe instead of reset TimeToWait replace this with a call to update?